services/brig/src/Brig/API/User.hs

```diff
@@ -257,8 +257,8 @@ createUser new@NewUser{..} = do
         ok <- lift $ Data.claimKey uk uid
         unless ok $
             throwE $ DuplicateUserKey uk
-        let minvmeta :: Maybe (UserId, UTCTimeMillis)
-            minvmeta = (, inCreatedAt inv) <$> inCreatedBy inv
+        let minvmeta :: (Maybe (UserId, UTCTimeMillis), Team.Permissions)
```

Comment:
userid is not always applicable (e.g. SCIM), so that *has* to be optional.

time and userid have just turned up together like that.  that could probably be changed, but that would increase the metadata storage further, and i don't see any advantage.

permissions are decided during creation of the `Invitation`.  it's likely i could keep the `Maybe Role` here and decide it in galley during team user creation, but why?  (i guess this has to do with your idea about admins getting removed from the system and the posthumusly causing member creations, but i'm still not entirely convinced.)